### PR TITLE
Add unit tests #5

### PR DESCRIPTION
**Pull request by jaredkerim**

### CODE REVIEW

File: embedly/embedly.py [hunk 1]
```diff
@@ -27,27 +29,29 @@ def get_cached_url(url):
...
         'urls={}'.format(','.join([urllib.quote_plus(url) for url in urls])),
     ])
 
-    request_url = '{base}?{params}'.format(base=EMBEDLY_URL, params=params)
+    request_url = '{base}?{params}'.format(
+        base=app.config['EMBEDLY_URL'],
+        params=params,
+    )
+
     response = requests.get(request_url)
```
Comment (ncloudioj) [inline embedly/embedly.py]: Better capture possible exceptions raised by requests.get, for example, `Timeout`, `ConnectionError`, and `TooManyRedirects`. Or capture `equests.exceptions.RequestException` to bail.

**Comment by ncloudioj** [comment]
@jaredkerim LGTM except one nit.

Since the unit tests are ready and more features will be added soon, I think it's time to turn on TravisCI/CircleCI and Coveralls now. What do you think?

**Comment by jaredkerim** [comment]
@ncloudioj Yep that is covered by #6!  Thanks for the review!

File: embedly/tests.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        mock_response = mock.Mock()
+        mock_response.content = json.dumps(embedly_data)
+        self.mock_requests_get.return_value = mock_response
+
+        response = self.client.get(self._build_query_url(self.sample_urls))
+
+        self.assertEqual(response.status_code, 200)
+
+        self.assertEqual(self.mock_redis.get.call_count, 2)
+        self.assertEqual(self.mock_requests_get.call_count, 1)
```
Comment (jaredkerim) [inline embedly/tests.py]: I should also check here that embedly was only called with the uncached URL.

File: embedly/tests.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    def _get_url_data(self, url):
+        return {
+            'original_url': url,
+        }
+
+    def _get_urls_data(self, urls):
+        return [self._get_url_data(url) for url in urls]
+
+    def _build_query_url(self, urls):
```
Comment (jaredkerim) [inline embedly/tests.py]: This should be extracted into a function in embedly.py and used by both the calling request handler and by the test.

Comment (jaredkerim) [inline embedly/tests.py]: D'oh, this is building the URL to request our proxy api, the one in embedly.py is for requesting the embedly API, two different call signatures.  Disregard this.

**Pull request merged**